specification/dsdl/serializable_types.tex

```diff
@@ -281,51 +313,282 @@ \subsubsection{Dependencies}
+
+\subsubsection{Extent and finality}\label{sec:dsdl_composite_extent_and_finality}
+
+As detailed in section \ref{sec:dsdl_versioning},
+data types may evolve over time to accommodate new design requirements, new features, to rectify issues, etc.
+In order to allow gradual migration of deployed systems to revised data types,
+it is desirable to ensure that they can be modified in a way that does not render new definitions
+incompatible with their earlier versions.
+In this context there are two related concepts:
+
+\begin{description}
+    \item[Extent] --- the minimum amount of memory, in bits, that shall be allocated to store the serialized
+    representation of the type.
+    The extent of any type is always greater than or equal the maximal value of its bit length set.
+    It is always a multiple of 8.
+
+    \item[Finality] --- denotes whether a type is \emph{final}.
+    A type that is final is non-evolvable and its extent equals the maximal value of its bit length set\footnote{%
+        I.e., the smallest possible extent.
+    }.
+    A type that is not final is also referred to as \emph{delimited}.
+\end{description}
+
+The extent is the growth limit for the maximal bit length of the type as it evolves.
+The extent should be at least as large as the longest serialized representation of any compatible version of the type,
+which ensures that an agent leveraging a particular version can correctly process any other compatible
+version due to the avoidance of unintended truncation of serialized representations.
+
+Serialized representations of evolvable definitions carry additional metadata which introduces a certain overhead.
+This may be undesirable in some scenarios, especially in cases where serialized representations of the
+definition are expected to be highly compact, thereby making the overhead comparatively more significant.
+In such cases, the designer may opt out of the extensibility by declaring the definition final.
+Serialized representations of final definitions do not incur the aforementioned overhead.
+
+The related mechanics are described in section \ref{sec:dsdl_serialization_composite_non_final}.
+
+\begin{figure}[H]
+    $$
+    \overbrace{%
+        \underbrace{%
+            \blacksquare\blacksquare\blacksquare\blacksquare\blacksquare\blacksquare%
+            \blacksquare\blacksquare\blacksquare\blacksquare\blacksquare\blacksquare%
+        }_{\substack{\text{Longest serialized} \\ \text{representation}}}%
+        \underbrace{%
+            \boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes%
+        }_{\substack{\text{Memory reserve} \\ \text{(none if final)}}}%
+    }^{\substack{%
+        \text{Extent} \\
+        \text{(memory requirement)}
+    }}
+    $$
+    \caption{Serialized representation and extent\label{fig:dsdl_extent}}
+\end{figure}
+
+The default extent of a non-final (delimited) definition is the following function
```

Comment:
You're going to have to give the reader a spoiler at this point that there are an additional 32-bits hiding in here as the delimiter header. Without that called out explicitly you find yourself utterly confused at your assertion that the 64 bit structure in your example (for example) takes up 96 bits.

Reply:
The delimiter header is not involved here. The 96 bits are explained right under the example:

<img width="560" alt="image" src="https://user-images.githubusercontent.com/3298404/95513469-145f6180-09c3-11eb-80cd-aab794ffb0ac.png">